src/js/select2/data/ajax.js

```diff
@@ -88,7 +88,7 @@ define([
       self._request = $request;
     }
 
-    if (this.ajaxOptions.delay && params.term !== '') {
+    if (this.ajaxOptions.delay && params.term !== undefined) {
```

Comment:
What's the use case for `''`? The input box is initialized as `undefined`, so avoiding the delay on initiation seems to be the desire here. Avoiding it on empty string causes the issue I mentioned originally.